### PR TITLE
Add file validation

### LINKED ISSUE #369: Validate files
Currently there is no validation of what files are available beyond checking that the epochs directories exist. There are two issues here:

1. When running on a system with an incomplete set of files, it becomes very difficult to troubleshoot which files do not exist. Additionally, there is no easy way to remove the search for specific files - the best way is to restrict the epochs that are queried, but that means that you lose all information from that epoch
2. The checks are done in the `Query` initialisation. As such, if users do not have every epoch available on their systems then they cannot run a `find_sources --find-fields` search without creating dummy directories.

I propose writing a `validate_files` function that is called inside `Query.find_sources` after the filenames have been added to the `sources_df`

### PR DESCRIPTION
**Pull request by ddobie**
Fix #369.

This PR checks that the requested files actually exist before proceeding with the query. Previously the query would run and inevitably break with an error message that was not at all clear to the user, or in some cases (e.g. if it is inside a dask compute job) exit with no error message.

### CODE REVIEW

**Comment by ddobie** [comment]
Not sure why @pep8speaks hasn't updated its comment, but running pycodestyle on my own machine shows that the code is PEP8 compliant.

File: vasttools/query.py [hunk 1]
```diff
@@ -380,12 +380,17 @@ def __init__(
...
+        all_data_available = self._check_data_availability()
+        if all_data_available:
+            self.logger.info("All data available!")
+        else:
+            self.logger.critical(
+                "Not all requested data is available! See above for details."
+            )
+            self.logger.critical(
+                "Query will continue run, but proceed with caution."
+            )
```
Comment (ajstewart) [inline vasttools/query.py:393]: I don't see this as a `critical` error - that usually denotes something that will cause the task to completely crash out. Given that the idea is that this picks out the missing ones and allows the query to continue it should probably be at the `warning` level?
```suggestion
            self.logger.warning(
                "Not all requested data is available! See above for details."
            )
            self.logger.warning(
                "Query will continue run, but proceed with caution."
            )
```

File: tests/test_query.py [hunk 2]
```diff
@@ -799,23 +799,15 @@ def test__check_data_availability(self,
...
-        with expectation as e:
-            query = vtq.Query(
-                epochs=epoch,
-                planets=['Mars'],
-                base_folder=base_dir,
-                stokes=stokes,
-                no_rms=no_rms
-            )
-            assert str(e) == message
+        assert all_available == query._check_data_availability()
```
Comment (ajstewart) [inline tests/test_query.py:810]: Can a test cover the `False` return case as well?

**Review by ajstewart — COMMENTED** [review]
Looks like a good change - I ran into this exact problem when trying to test the `find_fields` function locally!

Only a couple of comments on the test and the error level.

File: tests/test_query.py (hunk 2 above)
Comment (ddobie) [reply tests/test_query.py:810]: It does, doesn't it?

It's parametrised to cover a variety of scenarios

Comment (ajstewart) [reply tests/test_query.py:810]: Apologies I managed to completely miss that!

**Review by ajstewart — APPROVED** [review]

**Review by ajstewart — CHANGES_REQUESTED** [review]
Actually just realised - change log. Does any doc also need to be updated to state this behaviour?

**Comment by ddobie** [comment]
Good catch. Have updated.

**Review by ajstewart — APPROVED** [review]
Looks good to me!

**Pull request merged**